cmake/modules/boards.cmake

```diff
@@ -154,4 +154,6 @@ if(NOT BOARD_DIR)
   message(FATAL_ERROR "Invalid BOARD; see above.")
 endif()
 
-add_custom_target(boards ${list_boards_commands} USES_TERMINAL)
+if(NOT CMAKE_SCRIPT_MODE_FILE)
+  add_custom_target(boards ${list_boards_commands} USES_TERMINAL)
+endif()
```

Comment:
Should we document the desired outcome when boards.cmake is executed in script mode?

And perhaps mention in a comment in this file that it is supported to do so?

Otherwise we might break the intended behaviour by accident.

Reply:
It should.

Long term, then we could even consider to remove support for `ninja boards`.

After all, user has successfully generated a build system, so what's the real value in having a dedicated target that can list all boards ?

@mbolivar-nordic what do you think, is it time to remove `ninja boards` and go purely for the python scripts ?
This should probably go hand-in-hand with a `west shields` / `scripts/list_shields.py` script.

Reply:
@SebastianBoe I decided to override the `add_custom_target()` / `set_target_properties()` in the script to silence the CMake errors.

This removes the need for special handling / testing for `CMAKE_SCRIPT_MODE_FILE` at multiple locations in Zephyr CMake modules because the new approach is localized to the tooling script.
It also makes the commenting clearer.
https://github.com/zephyrproject-rtos/zephyr/blob/184940230028729634828431073444cd97733874/cmake/package_tooling.cmake#L50-L62

this allows us to write Zephyr CMake modules as we would normally do, and do the special handling in the code that wants to run only a subset of the build system.